ddtrace/propagation/http.py

```diff
@@ -27,7 +30,23 @@
 POSSIBLE_HTTP_HEADER_ORIGIN = frozenset([HTTP_HEADER_ORIGIN, get_wsgi_header(HTTP_HEADER_ORIGIN).lower()])
 
 
-class HTTPPropagator(object):
+class Propagator(six.with_metaclass(abc.ABCMeta)):
+    """Base Propagator using request headers as carrier."""
+
+    @staticmethod
+    @abc.abstractmethod
+    def inject(span_context, headers):
+        # type: (Context, Dict[str, str]) -> None
+        pass
+
+    @staticmethod
+    @abc.abstractmethod
+    def extract(headers):
```

Comment:
```suggestion
    def extract(*args, **kwargs):
```

I don't think we want to restrict the API of `extract`. (Same would be true for `inject`)